agimus_controller/agimus_controller/ocp_base_croco.py

```diff
@@ -0,0 +1,145 @@
+from abc import abstractmethod
+
+import crocoddyl
+import mim_solvers
+import numpy as np
+import numpy.typing as npt
+
+from agimus_controller.factory.robot_model import RobotModels
+from agimus_controller.mpc_data import OCPResults, OCPDebugData
+from agimus_controller.ocp_base import OCPBase
+from agimus_controller.ocp_param_base import OCPParamsBaseCroco
+from agimus_controller.trajectory import TrajectoryPointWeights
+
+
+class OCPBaseCroco(OCPBase):
+    def __init__(
+        self,
+        robot_models: RobotModels,
+        ocp_params: OCPParamsBaseCroco,
+    ) -> None:
+        """Defines common behavior for all OCP using croccodyl. This is an abstract class with some helpers to design OCPs in a more friendly way.
+
+        Args:
+            robot_models (RobotModels): All models of the robot.
+            ocp_params (OCPParamsBaseCroco): Input data structure of the OCP.
+        """
+        # Setting the robot model
+        self._robot_models = robot_models
+        self._collision_model = self._robot_models.collision_model
+        self._armature = self._robot_models._params.armature
+
+        # Stat and actuation model
+        self._state = crocoddyl.StateMultibody(self._robot_models.robot_model)
+        self._actuation = crocoddyl.ActuationModelFull(self._state)
+
+        # Setting the OCP parameters
+        self._ocp_params = ocp_params
+        self._ocp = None
+        self._ocp_results = None
+        self._running_model_list = None
```

Comment:
These are lists hence:
```suggestion
        self._running_model_list = []
```